tests/extendrtests/tests/testthat/test-ndarray.R

```diff
@@ -0,0 +1,12 @@
+test_that("[ndarray] Euclidean distance", {
+    m <- matrix(1.0 * 1:6, nrow = 3, ncol = 2)
+    dist <- euclidean_dist(m)
+
+    expect_equal(dist, c(sqrt(2), 2 * sqrt(2), sqrt(2)))
```

Comment:
Could you compare to R's native `dist(x, method = "euclidean")`?

Reply:
Good idea, I actually increased the size of the test matrix and randomized test input. Seems to work fine